src/ScriptCs.Hosting/Package/PackageObject.cs

```diff
@@ -60,6 +62,28 @@ public IEnumerable<string> GetCompatibleDlls(FrameworkName frameworkName)
             IEnumerable<IPackageFile> compatibleFiles;
             VersionUtility.TryGetCompatibleItems(frameworkName, dlls, out compatibleFiles);
 
+            // HACK: Delete unnecessary  temporary NuGet files
+            List<string> tempDirectories = new List<string>();
+            foreach (PhysicalPackageFile file in dlls)
+            {
+                var match = Regex.Match(((PhysicalPackageFile)dlls.First()).SourcePath, "(" + Path.Combine(Path.GetTempPath(), "nuget").Replace("\\", "\\\\") + "\\\\[^\\\\]+?)\\\\");
+                if (match.Success && match.Groups.Count > 1)
+                {
+                    tempDirectories.Add(match.Groups[1].Value); 
+                }
+
+            }
+
+            foreach (string directory in tempDirectories.Distinct())
+            {
+                if (Directory.Exists(directory))
+                {
+                    Directory.Delete(directory, true);
+                }
+            }
+
+            // /HACK
+
```

Comment:
Whilst acknowledging that this is always going to be a hack, my first question is whether this is the right place to put it.

This method retreives a list of the DLLs contained within the package which are compatible with the specified framework. Is this the right place to add the side effect of removing the folders from the temp folder? If the preceding (existing) lines in the method have the side effect of creating them, then it is the right place. If not, I think it may be worth searching for a better place to put this.